swag/swag_client.py

```diff
@@ -369,3 +416,294 @@ async def execute_swagdmin_command(self, message):
+            user_account = self.swag_bank.get_account_info(user.id)
+
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            history = list(reversed(self.swag_bank.get_cagnotte_history(cagnotte_idx)))
+
+            cagnotte_info = self.swag_bank.get_active_cagnotte(cagnotte_idx).get_info()
+            await message.channel.send(
+                f"{message.author.mention}, voici l'historique de tes transactions de la cagnotte **{cagnotte_info.name}** :\n"
+            )
+            await reaction_message_building(
+                self.client,
+                history,
+                message,
+                mini_history_swag_message,
+                self.swag_bank,
+                user_account.timezone,
+            )
+
+        elif "payer" in splited_command:
+
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            cagnotte = self.swag_bank.get_active_cagnotte(cagnotte_idx)
+
+            if cagnotte.currency == Currency.SWAG:
+                try:
+                    value = int(
+                        "".join(
+                            argent for argent in splited_command if argent.isnumeric()
+                        )
+                    )
+                except ValueError:
+                    raise InvalidSwagValue
+
+            elif cagnotte.currency == Currency.STYLE:
+                try:
+                    value = Decimal(
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.pay_to_cagnotte(message.author.id, cagnotte_idx, value)
+
+            cagnotte_info = cagnotte.get_info()
+            await message.channel.send(
+                "Transaction effectuée avec succès ! \n"
+                "```ini\n"
+                f"[{message.author.display_name}\t"
+                f"{format_number(value)} {currency_to_str(cagnotte_info.currency)}\t"
+                f"-->\t€{cagnotte_idx} {cagnotte_info.name}]\n"
+                "```"
+            )
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "donner" in splited_command:
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            cagnotte = self.swag_bank.get_active_cagnotte(cagnotte_idx)
+            destinataire = message.mentions
+            if len(destinataire) != 1:
+                await message.channel.send(
+                    "Merci de mentionner un destinataire (@Bobby Machin) pour "
+                    "lui donner une partie de cette €agnotte !"
+                )
+                return
+            destinataire = destinataire[0]
```

Comment:
`receiver`

Reply:
Resolved in : https://github.com/BobbySoftwares/Bybbot/pull/7/commits/c9cfecdb424780b7243e2b2ba8fcc0fba8a01e8f